wikimedia/wikipedia-ios
# Locally save native donations

**Phabricator:** https://phabricator.wikimedia.org/T372343

### Notes
* This PR adds logic and some UI  for saving locally a history of Apple Pay donations
* It doesn't address web donations
* I created a new `SharedContainerCacheRemover` that doesn't require a specific value, to allow for the whole directory deletion. I wanted to bypass the need to pass in a value <T> when not needed - in this case, I'm going for the whole file. Let me know if there's a better way to handle this case.

* Note: pending Design review for now - there was no Figma for this, so I created a simple Ui, and it's now under design review, but the saving and deleting logic can be reviewed independently of the UI code

### Test Steps
1. Make a donation 
2. Go to Settings -> Account -> Donation History
3. Make sure you see your donation with date and amount
4. Click delete.
5. You should see an empty view with a message
6. Background and terminate the app. Go to donation history again. Make sure the history is still empty
7. You can check the local storage, look for Donor Experience/AppLocalDonationHistory.json
